base/sort.jl

```diff
@@ -285,14 +285,14 @@ function searchsortedfirst(a::AbstractRange{<:Integer}, x::Real, o::DirectOrderi
         lastindex(a) + 1
     else
         if o isa ForwardOrdering
-            -fld(floor(Integer, -x) + first(a), h) + 1
+            -fld(floor(Integer, -x) + Signed(first(a)), h) + 1
```

Comment:
So the `Signed` here is for the case where `x` is not an `Integer` ? Just by curiosity, without this conversion here, the type conversion attached to the return value of the function (`::keytype(a)`) is not enough?

Reply:
No, it is for the case when `x` is `Signed` and `first(a)` is `Unsigned`. For example, if `floor(Integer, -x) == Int64(-5)` and `first(a) == UInt64(1)`, adding them yields the huge number `0xfffffffffffffffc` (the negative `Int64` gets promoted to `UInt64`). The result of the complete line is then a `UInt64` that is too large to convert to `Int`.

Reply:
Ah right I was off as `Signed` is not applied to `x`, and ok, the `::keytype(a)` fails then, thanks.